awscli/customizations/paginate.py

```diff
@@ -183,6 +184,23 @@ def check_should_enable_pagination(input_tokens, shadowed_args, argument_table,
             # what we're doing here.
             for key, value in shadowed_args.items():
                 argument_table[key] = value
+    
+    if not parsed_globals.paginate:
+        ensure_paging_params_not_set(parsed_args, shadowed_args)
+
+
+def ensure_paging_params_not_set(parsed_args, shadowed_args):
+    paging_params = ['starting_token', 'page_size', 'max_items']
+    shadowed_params = [p.replace('-', '_') for p in shadowed_args.keys()]
+    params_used = [p for p in paging_params if
+                   p not in shadowed_params and getattr(parsed_args, p)]
+
+    if len(params_used) > 0:
+        converted_params = ', '.join(
+            ["--" + p.replace('_', '-') for p in params_used])
+        raise PaginationError(
+            message="--no-paginate provided along with pagination "
```

Comment:
We can make the message a tad more explicit such as ``Cannot specify --no-paginate along with pagination arguments: ...``?